hub/src/server/authentication.ts

```diff
@@ -308,44 +369,54 @@ export class V1Authentication {
                                 ` not found in ${JSON.stringify(challengeTexts)}`)
     }
 
-    const expiresAt = decodedToken.payload.exp
+    const expiresAt = payload.exp
     if (expiresAt && expiresAt < (Date.now()/1000)) {
       throw new ValidationError(
         `Expired authentication token: expire time of ${expiresAt} (secs since epoch)`)
     }
 
-    if (decodedToken.payload.hasOwnProperty('associationToken') &&
-        decodedToken.payload.associationToken) {
+    if (payload.hasOwnProperty('associationToken') &&
+        payload.associationToken) {
       return this.checkAssociationToken(
-        decodedToken.payload.associationToken, address)
+        payload.associationToken, address)
     } else {
       return address
     }
   }
 }
 
-export class LegacyAuthentication {
-  publickey: bitcoin.ECPair
+export class LegacyAuthentication implements AuthenticationInterface {
+
+  checkAssociationToken(_token: string, _bearerAddress: string): void {
+    throw new Error('Method not implemented.')
+  }
+
+  parseAuthScopes(): AuthScopeValues {
+    // TODO: can legacy auth tokens be maliciously created to get around restrictive auth scopes?
```

Comment:
We need to know the answer to this before we merge!

Reply:
The answer needs to be a definitive "no" -- legacy auth tokens must be assigned a restricted set of scopes that the caller would have expected to receive (e.g. the ability to store data, but not delete or list data or add to collections).

Reply:
Sounds good.  We should remove this comment then.

Reply:
Removed comment